### DISCUSSION TITLE
Accessing Iventree from different devices

### DISCUSSION THREAD

**Discussion by ATLAS2246** [inventree/InvenTree#7402] [Q&A] [answered]
Im trying to setup inventree so i can access it from devices other than the one its hosting it, yet still on the same network. ive looked through the documentaton and havent had any luck understanding it, let alone setting it up. i have the database up and running on a rasberry pi and its working great so far but i wanted to see if someone could help me out or point me in the right direction.

Any help would be awesome!

**Comment by SchrodingersGat** [comment]
Assuming you have followed the docker setup guide, it is hosting on `http://inventree.localhost`, which is only accessible from that machine.

You will have to change `INVENTREE_SITE_URL` to the IP address of the host

![image](https://github.com/inventree/InvenTree/assets/10080325/f086ab3c-5fab-4b76-b2a7-e5fff05cf3ba)

**Reply by ATLAS2246** [reply]
i am using the docker setup, sorry i should have mentioned that. im assuming i needed to change it to show the internal ip address, 192.168 etc. ive tried that and the broadcast ip address yet when i try to go to that ip using chrome on my laptop it comes up with it shows "This site cannot provide a secure connection" is there something else i need to add to the url or caddy file which ive put here

```
`# Example Caddyfile for Inventree
'# The following environment variables may be used:
'# - INVENTREE_SITE_URL: The upstream URL of the Inventree site (default: 192.168.1.30)
'# - INVENTREE_SERVER: The internal URL of the Inventree container (default: http://inventree-server:8000)
'#
'# Note that while this file is a good starting point, it may need to be modified to suit your specific requirements


(log_common) {
	log {
		output file /var/log/caddy/{args[0]}.access.log
	}
}

(cors-headers) {
	header Allow GET,HEAD,OPTIONS
	header Access-Control-Allow-Origin *
	header Access-Control-Allow-Methods GET,HEAD,OPTIONS
	header Access-Control-Allow-Headers Authorization,Content-Type,User-Agent

	@cors_preflight{args[0]} method OPTIONS

	handle @cors_preflight{args[0]} {
		respond "" 204
	}
}

'# Change the host to your domain (this will serve at inventree.localhost)
{$INVENTREE_SITE_URL:192.168.1.30} {
	import log_common inventree

	encode gzip

	request_body {
		max_size 100MB
	}

	handle_path /static/* {
		import cors-headers static

		root * /var/www/static
		file_server
	}

	handle_path /media/* {
		import cors-headers media

		root * /var/www/media
		file_server

		header Content-Disposition attachment

		forward_auth {$INVENTREE_SERVER:"http://inventree-server:8000"} {
			uri /auth/
		}
	}

	reverse_proxy {$INVENTREE_SERVER:"http://inventree-server:8000"}
}'
```
Sorry if this is a bit low level, im very new to this

**Reply by SchrodingersGat** [reply]
You do not need to edit the caddy file at all. Simply edit the `INVENTREE_SITE_URL` parameter in your configuration - https://docs.inventree.org/en/stable/start/config/#site-url

**Reply by ATLAS2246** [reply]
Ive updated site URL to be 192.168.1.30 which is the Ip address for the rasberry pi its being hosted on so it shows site_url: 'https://192.168.1.30' i was still getting the 'this site cannot provide a secure connection'. following that i tried setting it in the .env file with the same value. although im still getting the same issue

**Reply by ATLAS2246** [reply]
sorry to bug you again, ive reinstalled through the docker intall again due to a different issue i ran into. would changing the site url to http://192.168.1.24:8000 be what i needed to do

**Reply by ATLAS2246** [reply]
Ive got it working, thanks for your help and patience with a beginner!

**Comment by Century1026** [comment]
maybe [this](https://github.com/inventree/InvenTree/discussions/7308#discussion-6716985) will help. see the marked answer

**Reply by ATLAS2246** [reply]
that looks like it will be helpful for stage 2 of this project lol. at the moment im just trying to make it accessable from a local network

**Reply by ATLAS2246** [reply]
Ive got it working, i ended up using this as a reference to better understand what was happening. Thanks!!

**Comment by ATLAS2246** [comment, marked as answer]
I've found a solution 

by modifying the .env file to show http://192.168.1.etc im able to access the database from other devices on the same network

I also have that modified in the config file under site_url, in the docs it shows that shouldnt matter but its working and im hesitant to change in back

ive also found it takes about a minute after restarting the server to be able to access the database over lan so that may have been causing me to think it wasnt working when it was

next step is to get it working over the internet!!

**Reply by gid204** [reply]
Hey @ATLAS2246 
I'm fighting with the config to get this working.

Are you able to please confirm exactly what files you edited and what you changed them to?

Currently I have in my .env file:
```
# Site URL - update this to match your host
#INVENTREE_SITE_URL="http://inventree.localhost"
INVENTREE_SITE_URL="http://192.168.78.168:8000"
```

and in my Cady config.yml (which I know I shouldn't change but am for now because you said you did):
```
# Base URL for the InvenTree server (or use the environment variable INVENTREE_SITE_URL)
#site_url: 'http://localhost:8000'
site_url: 'http://192.168.78.168:8000'
```

Thanks!

**Reply by gid204** [reply]
I was able to solve the issue by just adding these lines to the .env file:

```
# Site URL - update this to match your host
INVENTREE_SITE_URL="http://192.168.78.168"
INVENTREE_WEB_PORT=8000
```
I can now access InvenTree using another PC on my LAN by going to `http://192.168.78.168:8000`

With `192.168.78.168` being my server address.

**Reply by ATLAS2246** [reply]
Sorry I wasn't able to reply in time, but i'm glad you found the solution. in answer to the previous question I did end up having to re-deploy my instance to fix the changes I made to the caddy file.

**Comment by laenzlinger** [comment]
First of all, thank you very much for all the great work on InvenTree. I am a new user and was struggling with the same question. I was trying to just set the INVENTREE_SITE_URL.

I could not get it to work with this setting on http (not https).

To get it working on http only (insecure!!!) I also had to set INVENTREE_COOKIE_SAMESITE

I could not find this anywhere in the documentation. Therefore I wondered if there is something wrong with my docker-compose setup.

```
INVENTREE_SITE_URL="http://myhost.home" 
INVENTREE_COOKIE_SAMESITE=lax
```

**Reply by SchrodingersGat** [reply]
@laenzlinger have you tried updating to [version 0.16.7](https://github.com/inventree/InvenTree/releases) - there was a critical bug fix around the `INVENTREE_COOKIE_SAMESITE` variable. The new default value should be "false" - and it should work much better now. Please give that a try and see how you go.
